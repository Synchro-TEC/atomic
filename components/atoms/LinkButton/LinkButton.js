import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const LinkButton = ({ className, theme, onClick, style, children, ...otherProps }) => {
  const baseClass = 'sv-button link';
  const buttonTheme = `link-${theme}`;
  const rootClass = cx(baseClass, buttonTheme, className);

  return (
    <button className={rootClass} style={style} onClick={onClick} {...otherProps}>
      {children}
    </button>
  );
};

LinkButton.defaultProps = {
  theme: 'default',
};

LinkButton.propTypes = {
  theme: PropTypes.oneOf(['default', 'primary', 'info', 'danger', 'warning']),
  onClick: PropTypes.func,
  className: PropTypes.string,
  style: PropTypes.object,
};

export default LinkButton;
