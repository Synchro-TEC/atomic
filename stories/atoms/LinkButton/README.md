The `<LinkButton>` component is a button.