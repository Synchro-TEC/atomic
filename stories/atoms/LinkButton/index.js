import React from "react";
// import { text, boolean, object } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { optionalSelect } from "../../../utils/optionalSelect";

import readme from "./README.md";
import LinkButton from "../../../components/LinkButton";

const options = {
  default: "default",
  primary: "primary",
  info: "info",
  warning: "warning",
  danger: "danger",
};

const component = () => (
  <LinkButton
    theme={optionalSelect("Theme", options, "")}
    onClick={action("button_clicked")}
  >
    My Link Button
  </LinkButton>
);

export default [readme, component];