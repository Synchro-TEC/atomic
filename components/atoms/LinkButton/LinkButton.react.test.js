import React from 'react';
import LinkButton from './LinkButton';

describe('LinkButton Button Component', () => {
  test('Render the button', () => {
    const wrapper = shallow(<LinkButton>The Button</LinkButton>);
    expect(wrapper).toMatchSnapshot();
  });

  test('Fire onClick button', () => {
    const mockCallBack = jest.fn();
    const linkButton = shallow(<LinkButton onClick={mockCallBack}>The Button</LinkButton>);
    linkButton.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
