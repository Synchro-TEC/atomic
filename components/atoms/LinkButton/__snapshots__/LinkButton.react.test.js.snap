// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton Button Component Render the button 1`] = `
<button
  className="sv-button link link-default"
>
  The Button
</button>
`;
